src/EnergyPlus/HeatBalanceSurfaceManager.cc

```diff
@@ -571,15 +563,19 @@ namespace HeatBalanceSurfaceManager {
         // For daylit zones, calculate interior daylight illuminance at reference points and
         // simulate lighting control system to get overhead electric lighting reduction
         // factor due to daylighting.
-
-        for (SurfNum = 1; SurfNum <= TotSurfaces; ++SurfNum) {
-            if (Surface(SurfNum).Class == SurfaceClass_Window && Surface(SurfNum).ExtSolar) {
-                SurfaceWindow(SurfNum).IllumFromWinAtRefPtRep = 0.0;
-                SurfaceWindow(SurfNum).LumWinFromRefPtRep = 0.0;
+        for (int zoneNum = 1; zoneNum <= NumOfZones; ++zoneNum) {
+            int const firstSurfWin = Zone(zoneNum).WindowSurfaceFirst;
+            int const lastSurfWin = Zone(zoneNum).WindowSurfaceLast;
+            if (firstSurfWin == -1) continue;
```

Comment:
Hmm, these checks shouldn't be necessary. `WindowSurfaceFirst` is initialized to zero, and for zones with no windows, `WindowSurfaceLast` is [set to -1](https://github.com/NREL/EnergyPlus/blob/5ed2ba960287b86a57e58e320c075d65ccccd2f3/src/EnergyPlus/SurfaceGeometry.cc#L1834-L1837) any loops on windowsurfaces will be skipped over. In fact, looking at this now, `WindowSurfaceLast` and `NonWindowSurfaceLast` should simply be initialized to -1.

Reply:
Relying on windowSurfaceFirst to be initialized to 0 and windowSurfaceLast to be initialized to -1 is juuuuuuuust a little too cute. It's a recipe for bugs.

Reply:
@mjwitte I somehow wrongly pre-assumed this. And -1 is preferable if we want to convert these to vectors eventually? Can I make a follow-up PR to set `WindowSurfaceLast` and `NonWindowSurfaceLast` to -1 when no windows/opaqs and revised the checking?

Reply:
Whatever you do, I would keep the check. Don't get too cute with loop bounds. It's not worth it.

Reply:
@amirroth As it stand presently, `*First` and `*Last` are currently initialized to zero, and then `*Last` is set to -1 if there are no surfaces in that group. Is there a better way to skip those loops when they are empty?

Reply:
Yes, set them both to -1 by default (which works for both 1- and 0-based arrays) and do a separate check. A check before a loop whether to do the loop or not is fine, in fact the compiler often inserts checks like that itself. A check on every element in the loop is a lot less fine.